### PR TITLE
feat: Add `js_library` module 

### PR DESCRIPTION
**Pull request by Turbo87**
#### Description

This PR introduces a new `js_library` module, as suggested in https://github.com/starship/starship/pull/727#issuecomment-567644936

Under the hood it is checking all the specified libraries in parallel by using rayon and the [node-resolve](https://docs.rs/node-resolve/) crate to look for the corresponding `package.json` file.

By default the module is looking for `ember-source` and `react`.

#### Motivation and Context

> Why is this change required? What problem does it solve?

When working on multiple JS projects it is very useful to see at a glance what framework version is used by the project to figure out what features are available

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):

Bildschirmfoto 2019-12-13 um 09 30 24

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

I ran this locally via `target/debug/starship prompt` in several projects and I've included a few unit tests that should test the basic functionality.

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

I will update the documentation accordingly once the implementation is approved 🙏

### CODE REVIEW

**Comment by Turbo87** [comment]
@matchai as requested 😉

**Comment by Turbo87** [comment]
@matchai any thoughts on this implementation?

**Comment by matchai** [comment]
It, unfortunately, does look cluttered at this point.
What would be the ideal way to present all this information?

I was thinking that the Node version probably doesn't matter as much if there is a frontend library like Ember or React, but that wouldn't be the case if `js_library` was used for a backend library, like Express. 😕

**Comment by Turbo87** [comment]
@matchai Ember **apps** usually use `version: "0.0.0"` because they are usually not published to npm, which makes the package version display in starship somewhat unnecessary in that case. 

do you think hiding the package version if the `package.json` contains `private: true` or if the version is `0.0.0` is viable for starship?

File: tests/testsuite/js_library.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
+use ansi_term::Color;
```
Comment (matchai) [inline tests/testsuite/js_library.rs]: New module tests are now kept alongside the module they test (e.g. https://github.com/starship/starship/blob/master/src/modules/nodejs.rs#L38)

Comment (Turbo87) [reply tests/testsuite/js_library.rs]: no worries, I'll get it rebased and then move it around :)

**Comment by Turbo87** [comment]
unfortunately I currently don't have time to finish this. if anyone wants to take over feel free to rebase this :)

**Pull request closed without merging**